vars/buildPlugin.groovy

```diff
@@ -41,132 +41,140 @@ def call(Map params = [:]) {
+                            if (failFast && currentBuild.result == 'UNSTABLE') {
+                                error 'There were test failures; halting early'
                             }
-                            checkstyle(cp)
-                        }
-                        if (failFast && currentBuild.result == 'UNSTABLE') {
-                            error 'There were test failures; halting early'
-                        }
-                        if (doArchiveArtifacts) {
-                            if (incrementals) {
-                                String changelist = readFile(changelistF)
-                                dir(m2repo) {
-                                    fingerprint '**/*-rc*.*/*-rc*.*' // includes any incrementals consumed
-                                    archiveArtifacts artifacts: "**/*$changelist/*$changelist*",
-                                            excludes: '**/*.lastUpdated',
-                                            allowEmptyArchive: true // in case we forgot to reincrementalify
-                                }
-                                publishingIncrementals = true
-                            } else {
-                                String artifacts
-                                if (isMaven) {
-                                    artifacts = '**/target/*.hpi,**/target/*.jpi'
+                            if (doArchiveArtifacts) {
+                                if (incrementals) {
+                                    String changelist = readFile(changelistF)
+                                    dir(m2repo) {
+                                        fingerprint '**/*-rc*.*/*-rc*.*' // includes any incrementals consumed
+                                        archiveArtifacts artifacts: "**/*$changelist/*$changelist*",
+                                                excludes: '**/*.lastUpdated',
+                                                allowEmptyArchive: true // in case we forgot to reincrementalify
+                                    }
+                                    publishingIncrementals = true
                                 } else {
-                                    artifacts = '**/build/libs/*.hpi,**/build/libs/*.jpi'
+                                    String artifacts
+                                    if (isMaven) {
+                                        artifacts = '**/target/*.hpi,**/target/*.jpi'
+                                    } else {
+                                        artifacts = '**/build/libs/*.hpi,**/build/libs/*.jpi'
+                                    }
+                                    archiveArtifacts artifacts: artifacts, fingerprint: true
                                 }
-                                archiveArtifacts artifacts: artifacts, fingerprint: true
                             }
                         }
+                    }            
+                } finally {
```

Comment:
Probably does not really need a `finally` block, since most builds will pass and this is just a best effort anyway.

Reply:
does it do any harm though? should be more reliable?

Reply:
Only brought it up since it forced a lot of reindentation.